cli/src/cmd_auth.rs

```diff
@@ -267,84 +291,104 @@ impl CmdAuthLogin {
+        let host = self.host.as_host();
 
         // Construct a one-off API client, authenticated with the token
         // returned in the previous step, so that we can extract and save the
         // identity of the authenticated user.
-        let client = make_client(self.host.as_ref(), token.clone(), ctx.config());
+        let client = Client::new_authenticated_config(
+            &ClientConfig::default().with_host_and_token(host, &token),
+        )
+        .unwrap();
 
         let user = client.current_user_view().send().await?;
 
+        // TODO format this more reasonably
         println!("{:#?}", user);
 
+        let profile_name = profile.unwrap_or_else(|| &user.silo_name);
+
         // Set the user.
         // TODO: This should instead store the email, or some username or something
         // that is human knowable.
         // TODO ahl: not sure what even we're shooting for here. Maybe just a
         // way to understand the config files?
         let uid = user.id;
-        // ctx.config.set(host, "user", &email)?;
 
-        // Save the config.
-        // ctx.config.write()?;
+        println!("Logged in as {} under profile {}", uid, profile_name);
+
+        // Read / modify / write the credentials file.
+        let credentials_path = ctx.client_config().config_dir().join("credentials.toml");
+        let mut credentials =
+            if let Ok(contents) = std::fs::read_to_string(credentials_path.clone()) {
+                contents.parse::<toml_edit::DocumentMut>().unwrap()
+            } else {
+                Default::default()
+            };
 
-        println!("Logged in as {}", uid);
+        let profile_table = credentials
+            .entry("profile")
+            .or_insert_with(|| {
+                let mut table = Table::default();
+                // Avoid a bare [profile] table.
+                table.set_implicit(true);
+                Item::Table(table)
+            })
+            .as_table_mut()
+            .unwrap_or_else(|| {
+                panic!(
+                    "\"profile\" in {} is not a table",
+                    credentials_path.to_string_lossy()
+                )
+            });
+
+        let profile = profile_table
+            .entry(profile_name)
+            .or_insert_with(|| Item::Table(Table::default()))
+            .as_table_mut()
+            .unwrap_or_else(|| {
+                panic!(
+                    "\"profile.{}\" in {} is not a table",
+                    profile_name,
+                    credentials_path.to_string_lossy()
+                )
+            });
+
+        profile.insert("host", toml_edit::value(self.host.as_host().to_string()));
+        profile.insert("token", toml_edit::value(token));
+        profile.insert("user", toml_edit::value(uid.to_string()));
+
+        std::fs::write(credentials_path, credentials.to_string())
+            .expect("unable to write credentials.toml");
+
+        println!("{}", credentials);
```

Comment:
Could you add a comment at the start of the file warning users not to manually edit this file?